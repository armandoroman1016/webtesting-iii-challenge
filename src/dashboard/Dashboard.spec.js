// Test away
import React from 'react'
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react'

import Dashboard from './Dashboard'

describe ('<Dashboard />', () =>{
    it('matches snapshot', () => {
        const tree = renderer.create(<Dashboard />)
        expect(tree.toJSON()).toMatchSnapshot()
    })
    it('shows controls and display', () => {
        const { getByTestId } = render(<Dashboard />)
        const display = getByTestId(/display/i)
        const controls = getByTestId(/controls/i)
        expect(display).toBeTruthy();
        expect(controls).toBeTruthy();
    })
})