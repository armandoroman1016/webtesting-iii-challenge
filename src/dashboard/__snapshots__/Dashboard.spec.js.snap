// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> matches snapshot 1`] = `
Array [
  <div
    className="display panel"
    data-testid="display"
  >
    <div
      className="led green-led"
      data-testid="lockedStatus"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
      data-testid="openStatus"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
    data-testid="controls"
  >
    <button
      className="toggle-btn"
      data-testid="control-lock-unlock"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="control-open-close"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
